TS_HW1/home_work_1.ts

```diff
@@ -1,112 +1,121 @@
 
-  constructor(name, program) {
-    this.name = name;
+  constructor(name: string, program: string) {
+    this._name = name;
     this._program = program;
   }
 
-  get name() {
+  get name(): string {
     return this._name;
   }
 
-  get program() {
+  get program(): string {
     return this._program;
   }
 
-  addGroup(group) {
+  public addGroup(group: any) {
     this.groups.push(group);
   }
 }
 
 class Group {
-  _students = [];
-
-  get students() {
+  private _students: any[] = []; // : Student[]
+  public directionName: string;
+  public levelName: string;
+  
+  get students(): any[] {
     return this._students;
   }
 
-  constructor(directionName, levelName) {
+  constructor(directionName: string, levelName: string) {
     this.directionName = directionName;
     this.levelName = levelName;
   }
 
-  addStudent(student) {
+  public addStudent(student: any) {
     this._students.push(student);
   }
 
-  showPerformance() {
-    const sortedStudents = this.students.toSorted(
-      (a, b) => b.getPerformanceRating() - a.getPerformanceRating()
+  public showPerformance() {
+    const sortedStudents: any[] = this.students.toSorted(
+      (a: any, b: any) => b.getPerformanceRating() - a.getPerformanceRating()
     );
 
     return sortedStudents;
   }
 }
 
 class Student {
-  grades = {};
-  attendance = [];
-
-  constructor(firstName, lastName, birthYear) {
+  public grades: any = {};            // певний об'єкт Grade ?
+  public attendance: boolean[] = [];  // або об'єкт ?
+  public firstName: string;
+  public lastName: string;
+  public birthYear: number;
+  
+
+  constructor(firstName: string, lastName: string, birthYear: number) {
     this.firstName = firstName;
     this.lastName = lastName;
     this.birthYear = birthYear;
   }
 
-  get fullName() {
+  get fullName(): string {
     return `${this.lastName} ${this.firstName}`;
   }
 
-  set fullName(value) {
+  set fullName(value: string) {
     [this.lastName, this.firstName] = value.split(" ");
   }
 
-  get age() {
+  get age(): number {
     return new Date().getFullYear() - this.birthYear;
   }
 
-  setGrade(subject, grade) {
+  public setGrade(subject: string, grade: number): void {
     this.grades[subject] = grade;
   }
 
-  markAttendance(present) {
+  public markAttendance(present: boolean): void {
     this.attendance.push(present);
   }
 
-  getPerformanceRating() {
-    const gradeValues = Object.values(this.grades);
+  public getPerformanceRating(): number {
+    const gradeValues: number[] = Object.values(this.grades);
 
     if (gradeValues.length === 0) return 0;
 
-    const averageGrade =
-      gradeValues.reduce((sum, grade) => sum + grade, 0) / gradeValues.length;
+    const averageGrade: number =
+      gradeValues.reduce((sum: number, grade: number) => sum + grade, 0) / gradeValues.length;
 
-    const attendancePercentage =
+    const attendancePercentage: number =
       (this.attendance.filter((present) => present).length /
```

Comment:
You can explicitly add type annotation for filter variable:
`(present: boolean) => present`

It's something about code style. It's not a mistake for sure.